fl4health/clients/evaluate_client.py

```diff
@@ -157,21 +157,21 @@ def validate_on_model(
             for inputs, targets in self.data_loader:
                 inputs, targets = inputs.to(self.device), targets.to(self.device)
                 preds = {"prediction": model(inputs)}
-                losses = self.compute_loss(preds, {}, targets)
+                losses = self.compute_evaluation_loss(preds, {}, targets)
 
                 metric_meter.update(preds, targets)
                 loss_meter.update(losses)
 
         metrics = metric_meter.compute()
-        losses = loss_meter.compute()
+        losses = cast(EvaluationLosses, loss_meter.compute())
```

Comment:
If we do the generic typing refactor that I suggested below for the LossMeter class, I don't think this cast will be necessary, as .compute() will return the type T of the loss meter.

However, even if we don't do the generic typing refactor, I think we probably want to do
```
losses = loss_meter.compute()
assert isinstance(losses, EvaluationLosses)
```
rather than a `cast` for mypy. If I recall correctly, cast doesn't actually do any checking that the underlying object is actually of type EvaluationLosses. It essentially tells mypy to ignore any issues. The isinstance is a verification of the type and mypy infers the type from there.

I may be wrong though.